doc/rest_api/snippet_sharing/use_cases/UC_delete_snippet.txt

```diff
@@ -0,0 +1,17 @@
+Summary:
+----------------------------
+Title: delete configuration snippet
+Scope: Entry Management
+Level: User Goal
+Actors: Authenticated user, Moderator
+Brief: Users can delete their own database entries the same way that moderators can delete any configuration snippet in the database.
+
+Scenarios:
+----------------------------
+Precondition: Authenticated, sufficient permissions
+Main success scenario: Configuration snippet has been deleted successfully. A success response is returned.
+Alternative scenario: The specified snippet cannot be found (wrong URI) or the user has insufficient permissions to delete it. Either way the user is informed about the error by a response.
+Error scenario: Technical problems prevent the deletion process from completing. The user is informed by an error response.
+Postcondition: Configuration snippet has been deleted from the database and can neither be found through search, nor be viewed in detail anymore.

```

Comment:
Implementing it with git could be tricky if two deletes happen immediately after each other (what and how to roll back?), better would be to flag deleted resources and restore them. The problem with this approach is though that deleted resources should - after the delete - always return 404 not found for their URI. So how to properly restore them? In my opinion restoring them via create seems the best option, but I would not implement that in the API, but rather in the front-end (save the last deleted entry and automatically recreate it for undo).